global_shortcuts.go

```diff
@@ -0,0 +1,135 @@
+package astilectron
+
+import (
+	"context"
+	"sync"
+)
+
+const (
+	EventNameGlobalShortcutsCmdRegister          = "global.shortcuts.cmd.register"
+	EventNameGlobalShortcutsCmdIsRegistered      = "global.shortcuts.cmd.is.register"
+	EventNameGlobalShortcutsCmdUnregister        = "global.shortcuts.cmd.unregister"
+	EventNameGlobalShortcutsCmdUnregisterAll     = "global.shortcuts.cmd.unregister.all"
+	EventNameGlobalShortcutsEventRegistered      = "global.shortcuts.event.registered"
+	EventNameGlobalShortcutsEventIsRegistered    = "global.shortcuts.event.is.registered"
+	EventNameGlobalShortcutsEventUnregistered    = "global.shortcuts.event.unregistered"
+	EventNameGlobalShortcutsEventUnregisteredAll = "global.shortcuts.event.unregistered.all"
+	EventNameGlobalShortcutEventTriggered        = "global.shortcuts.event.triggered"
+)
+
+type globalShortcutsCallback func()
+
+// GlobalShortcuts represents a global shortcut
+type GlobalShortcuts struct {
+	*object
+	m         *sync.Mutex
+	callbacks map[string]*globalShortcutsCallback // Store all registered Global Shortcuts
+}
+
+func newGlobalShortcuts(ctx context.Context, d *dispatcher, i *identifier, w *writer) (gs *GlobalShortcuts) {
+
+	gs = &GlobalShortcuts{object: newObject(ctx, d, i, w, i.new()), m: new(sync.Mutex), callbacks: make(map[string]*globalShortcutsCallback)}
+	gs.On(EventNameGlobalShortcutEventTriggered, func(e Event) (deleteListener bool) { // Register the listener for the triggered event
+		gs.execCallback(e.GlobalShortcuts.Accelerator)
+		return
+	})
+	return
+}
+
+// Register Register global shortcuts
+func (gs *GlobalShortcuts) Register(accelerator string, callback globalShortcutsCallback) (isRegistered bool, err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to register the global shortcut
+	var event = Event{Name: EventNameGlobalShortcutsCmdRegister, TargetID: gs.id, GlobalShortcuts: &EventGlobalShortcuts{Accelerator: accelerator}}
```

Comment:
Could you not create a variable here and add the content of the event directly in `synchronousEvent`?

Also could you remove the empty line below?